src/ListValidator.php

```diff
@@ -25,13 +29,27 @@ public function doValidation( $value ) {
 			return;
 		}
 
-		$optionMap = array(
-			'elementcount' => 'range',
-			'maxelements' => 'upperbound',
-			'minelements' => 'lowerbound',
-		);
+		$validator = new RangeValidator();
+
+		if ( array_key_exists( 'elementcount', $this->options ) ) {
+			$range = $this->options['elementcount'];
+
+			if ( !is_array( $range ) || count( $range ) !== 2 ) {
+				throw new Exception( 'The elementcount option must be an array with two elements' );
```

Comment:
I did not see the old code and realize this was to match behaviour.

What I also somehow failed to notice is that the exception does not get thrown for an argument to this method, but for values in the options field, which makes InvalidArgumentException wrong. Could go for RuntimeException, though probably good to keep compat like you did before.

Sorry for the hassle, I should not have reviewed this when being tired :)